src/rgw/rgw_common.h

```diff
@@ -1917,6 +1934,7 @@ struct rgw_obj {
   rgw_obj_key key;
 
   bool in_extra_data{false}; /* in-memory only member, does not serialize */
+  bool head_obj{true}; /* in-memory only member, does not serialize */
```

Comment:
@fangyuxiangGL `rgw_obj` represents as it's name an rgw object. This means that it represents a set of both head and tail. This member implies that you aim to use it to point at only the head part, which doesn't make sense.